pkg/agent/endpoints/workload/handler_test.go

```diff
@@ -1515,13 +1516,14 @@ func runTest(t *testing.T, params testParams, fn func(ctx context.Context, clien
 		}),
 	))
 
+	sh := newStatsHandler()
 	server := grpc.NewServer(
 		grpc.UnaryInterceptor(unaryInterceptor),
 		grpc.StreamInterceptor(streamInterceptor),
+		grpc.StatsHandler(sh),
 	)
 	workloadPB.RegisterSpiffeWorkloadAPIServer(server, handler)
 	addr := spiretest.ServeGRPCServerOnTempUDSSocket(t, server)
-	t.Cleanup(func() { server.Stop() })
```

Comment:
This is already done by spiretest.ServeGRPCServerOnTempUDSSocket and is redundant.